sdk/storage/azure_storage_queue/tests/queue_client.rs

```diff
@@ -0,0 +1,653 @@
+        assert_successful_response(&response);
+
+        let message = response.into_body().await?;
+
+        assert!(
+            message.is_none(),
+            "Expected to receive no message, but got Some"
+        );
+
+        Ok::<(), azure_core::Error>(())
+    }
+    .await;
+
+    // Clean up by deleting the queue - this always executes
+    queue_client.delete(None).await.unwrap();
+
+    test_result
+}
+
+/// Peeks the first message from a queue in Azure Storage Queue service.
+#[recorded::test]
+async fn test_peek_message(ctx: TestContext) -> Result<()> {
+    let recording = ctx.recording();
+    let queue_client = get_queue_client(recording, "test-peek-message").await?;
+    let test_messages = ["Message 1", "Message 2"];
+
+    // Setup test queue with messages
+    setup_test_queue_with_messages(&queue_client, &test_messages).await?;
+
+    // Run the test logic and ensure cleanup always happens
+    let test_result = async {
+        peek_and_assert(&queue_client, &test_messages, 1, None).await?;
+
+        // The messages should not have been dequeued, so we can peek again
+        // and expect to receive the same message again.
+        peek_and_assert(&queue_client, &test_messages, 1, None).await?;
+
+        Ok(())
+    }
+    .await;
+
+    // Clean up by deleting the queue - this always executes
+    queue_client.delete(None).await.unwrap();
+
+    test_result
+}
+
+/// Receives all messages from a queue in Azure Storage Queue service.
+#[recorded::test]
+async fn test_peek_messages(ctx: TestContext) -> Result<()> {
+    let recording = ctx.recording();
+    let queue_client = get_queue_client(recording, "test-peek-messages").await?;
+    let test_messages = ["Message 1", "Message 2"];
+
+    // Setup test queue with messages
+    setup_test_queue_with_messages(&queue_client, &test_messages).await?;
+
+    // Run the test logic and ensure cleanup always happens
+    let test_result = async {
+        let options = Some(QueueClientPeekMessagesOptions {
+            number_of_messages: Some(10),
+            ..Default::default()
+        });
+
+        peek_and_assert(
+            &queue_client,
+            &test_messages,
+            test_messages.len(),
+            options.clone(),
+        )
+        .await?;
+
+        // The messages should not have been dequeued, so we can peek again
+        // and expect to receive both messages this time.
+        peek_and_assert(
+            &queue_client,
+            &test_messages,
+            test_messages.len(),
+            options.clone(),
+        )
+        .await?;
+
+        Ok(())
+    }
+    .await;
+
+    // Clean up by deleting the queue - this always executes
+    queue_client.delete(None).await.unwrap();
+
+    test_result
+}
+
+/// Attempts to receive the first message from an empty queue in Azure Storage Queue service.
+#[recorded::test]
+async fn test_dequeue_message_empty(ctx: TestContext) -> Result<()> {
```

Comment:
Let's rename the dequeue test to match the APi, `receive_message`.

Please note if you rename or delete tests, its best to manually remove the old recordings from the assets. Unfortunately, this does not happen automatically.